### DISCUSSION TITLE
CustomStatusBuilder

### DISCUSSION THREAD

**Discussion by skllll06** [flyerhq/flutter_chat_ui#281] [Ideas & Requests]
Hi,
first of thanks for developing this great package!!

I am very happy with it but I do think it can be improved in status builder.
We may want to display the number of people who have read your messages, and their avatars like other chat apps.
I would be happy if I could freely pass the Widget with a message to the status Widget at that time!
The code then would be:

```dart:message.dart
  final Widget Function(types.Message message, {required BuildContext context})?
      customStatusBuilder;
```

```dart:message.dart
if (_currentUserIsAuthor)
  Padding(
    padding: InheritedChatTheme.of(context).theme.statusIconPadding,
    child: showStatus
        ? GestureDetector(
            onLongPress: () =>
                onMessageStatusLongPress?.call(context, message),
            onTap: () => onMessageStatusTap?.call(context, message),
            child: customStatusBuilder != null
                ? customStatusBuilder!(message, context: context) // add this code
                : _statusBuilder(context),
          )
        : null,
  ),
```

Best wishes.

**Comment by demchenkoalex** [comment]
Hey! Thanks for the kind words! Would you be open to make this a PR? I will merge it.

**Reply by skllll06** [reply]
@demchenkoalex 
i created a PR,
Thanks.
